pkg/scheduler/apis/config/v1beta2/defaults.go

```diff
@@ -0,0 +1,226 @@
+	return RegisterDefaults(scheme)
+}
+
+// SetDefaults_KubeSchedulerConfiguration sets additional defaults
+func SetDefaults_KubeSchedulerConfiguration(obj *v1beta2.KubeSchedulerConfiguration) {
+
+	if obj.Parallelism == nil {
+		obj.Parallelism = pointer.Int32Ptr(16)
+	}
+
+	if len(obj.Profiles) == 0 {
+		obj.Profiles = append(obj.Profiles, v1beta2.KubeSchedulerProfile{})
+	}
+	// Only apply a default scheduler name when there is a single profile.
+	// Validation will ensure that every profile has a non-empty unique name.
+	if len(obj.Profiles) == 1 && obj.Profiles[0].SchedulerName == nil {
+		obj.Profiles[0].SchedulerName = pointer.StringPtr(api.DefaultSchedulerName)
+	}
+
+	// For Healthz and Metrics bind addresses, we want to check:
+	// 1. If the value is nil, default to 0.0.0.0 and default scheduler port
+	// 2. If there is a value set, attempt to split it. If it's just a port (ie, ":1234"), default to 0.0.0.0 with that port
+	// 3. If splitting the value fails, check if the value is even a valid IP. If so, use that with the default port.
+	// Otherwise use the default bind address
+	defaultBindAddress := net.JoinHostPort("0.0.0.0", strconv.Itoa(config.DefaultInsecureSchedulerPort))
+	if obj.HealthzBindAddress == nil {
+		obj.HealthzBindAddress = &defaultBindAddress
+	} else {
+		if host, port, err := net.SplitHostPort(*obj.HealthzBindAddress); err == nil {
+			if len(host) == 0 {
+				host = "0.0.0.0"
+			}
+			hostPort := net.JoinHostPort(host, port)
+			obj.HealthzBindAddress = &hostPort
+		} else {
+			// Something went wrong splitting the host/port, could just be a missing port so check if the
+			// existing value is a valid IP address. If so, use that with the default scheduler port
+			if host := net.ParseIP(*obj.HealthzBindAddress); host != nil {
+				hostPort := net.JoinHostPort(*obj.HealthzBindAddress, strconv.Itoa(config.DefaultInsecureSchedulerPort))
+				obj.HealthzBindAddress = &hostPort
+			} else {
+				// TODO: in v1beta2 we should let this error instead of stomping with a default value
+				obj.HealthzBindAddress = &defaultBindAddress
+			}
+		}
+	}
+
+	if obj.MetricsBindAddress == nil {
+		obj.MetricsBindAddress = &defaultBindAddress
+	} else {
+		if host, port, err := net.SplitHostPort(*obj.MetricsBindAddress); err == nil {
+			if len(host) == 0 {
+				host = "0.0.0.0"
+			}
+			hostPort := net.JoinHostPort(host, port)
+			obj.MetricsBindAddress = &hostPort
+		} else {
+			// Something went wrong splitting the host/port, could just be a missing port so check if the
+			// existing value is a valid IP address. If so, use that with the default scheduler port
+			if host := net.ParseIP(*obj.MetricsBindAddress); host != nil {
+				hostPort := net.JoinHostPort(*obj.MetricsBindAddress, strconv.Itoa(config.DefaultInsecureSchedulerPort))
+				obj.MetricsBindAddress = &hostPort
+			} else {
+				// TODO: in v1beta2 we should let this error instead of stomping with a default value
+				obj.MetricsBindAddress = &defaultBindAddress
```

Comment:
isn't this the time to stop defaulting this and validate it? this is likely to get lost and ship in v1beta2 accidentally

Reply:
Done.